catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorObjectListTest.java

```diff
@@ -0,0 +1,168 @@
+
+	@Parameterized.Parameters(name = "{1}" + "-Test")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{R.string.formula_editor_object_transparency, "transparency"},
+				{R.string.formula_editor_object_brightness, "brightness"},
+				{R.string.formula_editor_object_color, "color"},
+				{R.string.formula_editor_object_background_number, "look_number"},
+				{R.string.formula_editor_object_background_name, "look_name"},
+				{R.string.formula_editor_object_x, "position_x"},
+				{R.string.formula_editor_object_y, "position_y"},
+				{R.string.formula_editor_object_size, "size"},
+				{R.string.formula_editor_object_rotation, "direction"},
+				{R.string.formula_editor_object_layer, "layer"},
+				{R.string.formula_editor_function_collides_with_edge, "touches_edge"},
+				{R.string.formula_editor_function_collision, "touches_object"},
+				{R.string.formula_editor_function_touched, "touches_finger"},
+				{R.string.formula_editor_object_x_velocity, "x_velocity"},
+				{R.string.formula_editor_object_y_velocity, "y_velocity"},
+				{R.string.formula_editor_object_angular_velocity, "angular_velocity"}
+
+		});
+	}
+
+	@Parameter
+	public @StringRes
+	int formulaEditorObjectFunction;
+
+	@Parameter(1)
+	public String testName;
+
+	@Before
+	public void setUp() throws Exception {
+		Project project = new Project(null, "FormulaEditorObjectTest");
+		Sprite sprite = new Sprite("testSprite");
+		Sprite sprite2 = new Sprite(nameSprite2);
+		Script script = new StartScript();
+
+		sprite.addScript(script);
+		project.getDefaultScene().addSprite(sprite);
+		project.getDefaultScene().addSprite(sprite2);
+
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+		script.addBrick(new ChangeSizeByNBrick(0));
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testLogicListElements() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_change_size_by);
+		onBrickAtPosition(1).perform(click());
+
+		onView(withText(R.string.formula_editor_object)).perform(click());
+
+		String editorFunction = InstrumentationRegistry.getTargetContext().getString(formulaEditorObjectFunction);
+
+		onData(allOf(is(instanceOf(String.class)),is(editorFunction)))
+				.inAdapterView(FormulaEditorFunctionListMatchers.isFunctionListView())
+				.check(matches(isDisplayed()));
+
+		onView(withText(editorFunction)).perform(click());
+
+		if (editorFunction.compareTo(getResources().getString(R.string.formula_editor_function_collision)) == 0){
+			onView(withText(nameSprite2)).check(matches(isDisplayed()));
+			onView(allOf(withText(R.string.ok), hasSibling(withText(R.string.cancel)))).perform(click());
+			onView(withId(R.id.formula_editor_edit_field))
```

Comment:
i really dont understand whats going on here with the sprite name... and why the test has a if/else branch...